comp/core/workloadmeta/collectors/util/ecs_util.go

```diff
@@ -0,0 +1,267 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build docker
+
+package util
+
+import (
+	"context"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-agent/comp/core/workloadmeta"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/ecs/metadata/v3or4"
+	"github.com/DataDog/datadog-agent/pkg/util/flavor"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// TaskParser is a function that parses ECS tasks
+type TaskParser func(ctx context.Context) ([]workloadmeta.CollectorEvent, error)
+
+// IsTaskCollectionEnabled returns true if the task metadata collection is enabled for core agent
+// If agent launch type is EC2, collector will query the latest ECS metadata endpoint for each task returned by v1/tasks
+// If agent launch type is Fargate, collector will query the latest ECS metadata endpoint
+func IsTaskCollectionEnabled() bool {
```

Comment:
Following the other comment regarding using the config component I would suggest passing the config component to this function. That way we do not use `pkg/config` here either 😄

Reply:
https://github.com/DataDog/datadog-agent/pull/21836/commits/781a3e68bc0127ce695565d2010f617e53c05dfe